src/main/java/PasswordGenerator.java

```diff
@@ -22,18 +21,24 @@ public String createStringOfRandomCharacters() {
 
         String randomCharacters = "";
 
+        //This could be a for loop
         randomCharacters += pickCharactersFromList(1, characterTypeList.get(0));
         randomCharacters += pickCharactersFromList(1, characterTypeList.get(1));
         randomCharacters += pickCharactersFromList(1, characterTypeList.get(2));
 
         return randomCharacters;
     }
 
+    //method visibility?
     public String pickCharactersFromList(int numberAmount, String alphabet) {
+        //variable naming?
         int N = alphabet.length();
+        //maybe character isn't the best name?
         String character = "";
+        //resuse the Random instance?
         Random r = new Random();
 
+        //concatenating strings in a loop
         for (int i = 0; i < numberAmount; i++) {
            character += alphabet.charAt(r.nextInt(N));
```

Comment:
Given that we're dealing with short strings in this case it's no big deal but generally you should be wary of using the + operator inside loops, as it's creating a new string every time, which would be inefficient if the number of iterations was large.